libs/common/result.hpp

```diff
@@ -106,8 +107,73 @@ namespace iroha {
                               std::forward<ValueMatch>(value_func),
                               std::forward<ErrorMatch>(error_func));
       }
+
+      constexpr bool has_val() const {
+        return some_val();
+      }
+
+      constexpr bool has_err() const {
+        return some_err();
+      }
+
+      constexpr auto some_val() const {
```

Comment:
The naming is not obvious, maybe a comment can help?
Same for all new methods in the file except has_val, has_err